src/overlays/actors/ovl_En_Zog/z_en_zog.c

```diff
@@ -43,78 +48,1047 @@ const ActorInit En_Zog_InitVars = {
+        this->unk_320 = 5;
+        switch (globalCtx->msgCtx.unk11F04) {
+            case 0x1004:
+            case 0x1005:
+            case 0x1006:
+                func_80151938(globalCtx, globalCtx->msgCtx.unk11F04 + 1);
+                break;
+
+            case 0x1007:
+                func_801477B4(globalCtx);
+                this->actionFunc = func_80B94E34;
+                this->unk_300 = 5;
+                this->unk_320 = 0;
+                this->unk_31C = 2;
+                this->unk_31E = 0;
+                func_80B93B44(this);
+                break;
+        }
+    }
+    func_80B93A48(this, globalCtx);
+}
+
+void func_80B94E34(EnZog* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = GET_PLAYER(globalCtx);
+
+    func_80B93D2C(this, globalCtx);
+    func_80B93BE0(this, globalCtx);
+    if (this->actor.speedXZ < 0.1f) {
+        this->actor.speedXZ = 0.0f;
+    } else {
+        if (this->actor.speedXZ > 0.1f) {
+            WaterBox* sp44;
+            Vec3f sp38;
+
+            Lib_Vec3f_TranslateAndRotateY(&this->actor.world.pos, this->actor.shape.rot.y, &D_80B959AC, &sp38);
+            sp38.x += randPlusMinusPoint5Scaled(30.0f);
+            sp38.y += 20.0f;
+            sp38.z += randPlusMinusPoint5Scaled(30.0f);
+            if (func_800CA1AC(globalCtx, &globalCtx->colCtx, sp38.x, sp38.z, &sp38.y, &sp44) &&
+                (this->actor.world.pos.y < sp38.y)) {
+                EffectSsGSplash_Spawn(globalCtx, &sp38, NULL, NULL, 1,
+                                      Rand_ZeroFloat(this->actor.speedXZ * 40.0f) + (this->actor.speedXZ * 60.0f));
+            }
+
+            if ((player->actor.speedXZ > 3.0f) && (this->unk_324 == 0)) {
+                this->unk_324 = 25;
+                func_800B8E58(&player->actor, player->ageProperties->unk_92 + 0x6818);
+            }
+        }
+        this->actor.speedXZ *= 0.3f;
+    }
+
+    if (ABS_ALT(this->actor.yawTowardsPlayer - this->actor.world.rot.y) > 0x5000) {
+        func_800B8A1C(&this->actor, globalCtx, 0, 60.0f, 40.0f);
+    }
+
+    if (this->unk_324 > 0) {
+        this->unk_324--;
+    }
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = func_80B94D0C;
+        this->actor.speedXZ = 0.0f;
+        this->unk_300 = 2;
+    } else if (this->actor.bgCheckFlags & 1) {
+        this->actor.home.rot.z = 1;
+        this->actionFunc = func_80B94C5C;
+        this->actor.speedXZ = 0.0f;
+        this->unk_2FE = 1;
+        this->actor.velocity.y = 0.0f;
+        this->actor.minVelocityY = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->unk_31C = 1;
+        this->unk_31E = 0;
+        func_80B93BA8(this, 0);
+        gSaveContext.weekEventReg[88] |= 0x10;
+    } else {
+        if ((this->actor.yawTowardsPlayer > 16000) && (this->actor.yawTowardsPlayer < 32000) && (this->unk_302 == 0)) {
+            func_800B8614(&this->actor, globalCtx, 150.0f);
+        }
+    }
```

Comment:
You can merge those two into one conditional
```suggestion
    } else if ((this->actor.yawTowardsPlayer > 16000) && (this->actor.yawTowardsPlayer < 32000) && (this->unk_302 == 0)) {
        func_800B8614(&this->actor, globalCtx, 150.0f);
    }
```